src/content/datachannel/filetransfer/js/main.js

```diff
@@ -7,77 +8,76 @@
  */
 'use strict';
 
-var localConnection;
-var remoteConnection;
-var sendChannel;
-var receiveChannel;
-var bitrateDiv = document.querySelector('div#bitrate');
-var fileInput = document.querySelector('input#fileInput');
-var downloadAnchor = document.querySelector('a#download');
-var sendProgress = document.querySelector('progress#sendProgress');
-var receiveProgress = document.querySelector('progress#receiveProgress');
-var statusMessage = document.querySelector('span#status');
-
-var receiveBuffer = [];
-var receivedSize = 0;
-
-var bytesPrev = 0;
-var timestampPrev = 0;
-var timestampStart;
-var statsInterval = null;
-var bitrateMax = 0;
+let localConnection;
+let remoteConnection;
+let sendChannel;
+let receiveChannel;
+const bitrateDiv = document.querySelector('div#bitrate');
+const fileInput = document.querySelector('input#fileInput');
+const downloadAnchor = document.querySelector('a#download');
+const sendProgress = document.querySelector('progress#sendProgress');
+const receiveProgress = document.querySelector('progress#receiveProgress');
+const statusMessage = document.querySelector('span#status');
+
+let receiveBuffer = [];
+let receivedSize = 0;
+
+let bytesPrev = 0;
+let timestampPrev = 0;
+let timestampStart;
+let statsInterval = null;
+let bitrateMax = 0;
 
 fileInput.addEventListener('change', handleFileInputChange, false);
 
-function handleFileInputChange() {
-  var file = fileInput.files[0];
+async function handleFileInputChange() {
+  let file = fileInput.files[0];
   if (!file) {
-    trace('No file chosen');
+    console.log('No file chosen');
```

Comment:
While timestamps might be useful, I believe it is more useful to be able to track the line in the js file where this was printed. trace() is defined in a different file, so all these calls will look like it came from there.

Between timestamp and finding where the log was printed, I'd choose the later.

Reply:
Can we have our cake and eat it too?
ie change console.log(msg) to console.log(ts(), msg), where ts() returns a properly formatted timestamp?

Reply:
That would make things like; `console.log('Got this thing: ', anObject)` difficult to do. 

We could redefine `console.log`, but IMO the timestamp is not really needed. Everything happens on a single thread anyway, so it will be logged in chronological order. Not sure what the timestamp actually helps with in this context.